### PR TITLE
add Initialize() func to jsonreport

### PR DESCRIPTION
**Pull request by TrevorFarrelly**
**What this PR does, why we need it**:
Allows us to initialize Prow from the jsonreport library, bypassing a panic that occurs when `golang.org/x/net/trace.init()` is called from multiple contexts. Necessary for further development of the flaky-test-retryer.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes to reviewers**:

**User-visible changes in this PR**:

### CODE REVIEW

File: tools/flaky-test-reporter/jsonreport/jsonreport.go [hunk 1]
```diff
@@ -40,6 +40,10 @@ type Report struct {
 	Flaky []string `json:"flaky"`
 }
 
+func Initialize(credentials string) error {
+	return prow.Initialize(credentials)
```
Comment (chaodaiG) [inline tools/flaky-test-reporter/jsonreport/jsonreport.go]: suggest also creates a struct for handling everything jsonreport related, so the caller doesn't receive `prow` struct when call `Initialize`

Comment (TrevorFarrelly) [reply tools/flaky-test-reporter/jsonreport/jsonreport.go]: I don't believe there is a struct or a client in this case, unlike other setup functions we use. prow.Initialize() is just a wrapper for gcs.Authenticate(), and all structs are private within that package.

File: tools/flaky-test-reporter/jsonreport/jsonreport.go [hunk 2]
```diff
@@ -40,6 +40,7 @@ type Report struct {
 	Flaky []string `json:"flaky"`
 }
 
+// Initialize wraps prow's init, which must be called before any other prow functions are used.
 func Initialize(credentials string) error {
```
Comment (chaodaiG) [inline tools/flaky-test-reporter/jsonreport/jsonreport.go]: please rename `credentials` to `serviceAccount` for consistency, this is actually a file path afaik

**Pull request merged**